mlflow/sklearn/__init__.py

```diff
@@ -549,6 +881,39 @@ def autolog(
           .. _r2 score:
               https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html
 
+     **Post training metrics**
```

Comment:
@WeichenXu123 I left some minor suggestions / edits on these docs, but this is awesome! Great job writing this up!

cc @harupy

Reply:
all addressed.